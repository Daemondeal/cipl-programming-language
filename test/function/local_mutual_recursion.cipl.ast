pass
(block
    (proc is_even (params n) (body
        (if (== n 0) (then (block
            (return true)
        ))
        (return (call is_odd (- n 1)))
    )
    (proc is_odd (params n) (body
        (if (== n 0) (then (block
            (return false)
        ))
        (return (call is_even (- n 1)))
    )
    (expression (call print (call is_even 4)))
)

