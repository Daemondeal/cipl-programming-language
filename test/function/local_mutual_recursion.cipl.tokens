Token          Lexeme         Line 
PASS           pass           1    
LINE_END       \n             2    
INDENT                        2    
PROC           proc           2    
IDENTIFIER     is_even        2    
LEFT_PAREN     (              2    
IDENTIFIER     n              2    
RIGHT_PAREN    )              2    
COLON          :              2    
LINE_END       \n             3    
INDENT                        3    
IF             if             3    
IDENTIFIER     n              3    
EQUAL_EQUAL    ==             3    
NUMBER         0              3     0
COLON          :              3    
LINE_END       \n             4    
INDENT                        4    
RETURN         return         4    
TRUE           true           4    
LINE_END       \n             5    
DEDENT                        5    
RETURN         return         5    
IDENTIFIER     is_odd         5    
LEFT_PAREN     (              5    
IDENTIFIER     n              5    
MINUS          -              5    
NUMBER         1              5     1
RIGHT_PAREN    )              5    
LINE_END       \n             6    
DEDENT                        7    
PROC           proc           7    
IDENTIFIER     is_odd         7    
LEFT_PAREN     (              7    
IDENTIFIER     n              7    
RIGHT_PAREN    )              7    
COLON          :              7    
LINE_END       \n             8    
INDENT                        8    
IF             if             8    
IDENTIFIER     n              8    
EQUAL_EQUAL    ==             8    
NUMBER         0              8     0
COLON          :              8    
LINE_END       \n             9    
INDENT                        9    
RETURN         return         9    
FALSE          false          9    
LINE_END       \n             10   
DEDENT                        11   
RETURN         return         11   
IDENTIFIER     is_even        11   
LEFT_PAREN     (              11   
IDENTIFIER     n              11   
MINUS          -              11   
NUMBER         1              11    1
RIGHT_PAREN    )              11   
LINE_END       \n             12   
DEDENT                        13   
IDENTIFIER     print          13   
LEFT_PAREN     (              13   
IDENTIFIER     is_even        13   
LEFT_PAREN     (              13   
NUMBER         4              13    4
RIGHT_PAREN    )              13   
RIGHT_PAREN    )              13   
LINE_END       \n             14   
DEDENT                        14   
EOF                           14   
