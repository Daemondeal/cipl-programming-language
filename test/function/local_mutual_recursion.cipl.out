::stdout::

::stderr::
Error: Undefined variable 'is_odd'. at line 5
